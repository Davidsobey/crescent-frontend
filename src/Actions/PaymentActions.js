import PaymentConstants from '../Constants/PaymentConstants';
import PaymentService from '../Services/PaymentService';
import AlertActions from './AlertActions';
import history from '../Helpers/History';

function getPaymentStatuses() {
  function request() {
    return { type: PaymentConstants.GETALL_REQUEST };
  }
  function success(paymentStatuses) {
    return { type: PaymentConstants.GETALL_SUCCESS, paymentStatuses };
  }
  function failure(error) {
    return { type: PaymentConstants.GETALL_FAILURE, error };
  }

  return (dispatch) => {
    dispatch(request());

    PaymentService.getAll().then(
      paymentStatuses => dispatch(success(paymentStatuses)),
      (error) => {
        dispatch(failure(error));
        dispatch(AlertActions.error(error));
      },
    );
  };
}

function changePaymentStatus(clientId, courseId) {
  function request() {
    return { type: PaymentConstants.CHANGE_PAYMENT_STATUS_REQUEST };
  }
  function success() {
    return { type: PaymentConstants.CHANGE_PAYMENT_STATUS_SUCCESS };
  }
  function failure(error) {
    return { type: PaymentConstants.CHANGE_PAYMENT_STATUS_FAILURE, error };
  }

  return (dispatch) => {
    dispatch(request());

    PaymentService.changePaymentStatus(clientId, courseId).then(
      () => {
        history.push('/admin/home');
        dispatch(success());
        dispatch(AlertActions.success('Payment Status Updated'));
      },
      (error) => {
        dispatch(failure(error));
        dispatch(AlertActions.error(error));
      },
    );
  };
}
const PaymentActions = {
  getPaymentStatuses,
  changePaymentStatus,
};

export default PaymentActions;
