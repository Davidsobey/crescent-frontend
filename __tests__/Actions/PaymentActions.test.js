import configureStore from 'redux-mock-store';

// Actions to be tested
import PaymentActions from '../../src/Actions/PaymentActions';

const mockStore = configureStore();
const store = mockStore();

describe('PaymentActions', () => {
  beforeEach(() => { // Runs before each test in the suite
    store.clearActions();
  });

  test('Dispatches the correct action and payload', () => {
    expect(store.getActions()).toMatchSnapshot();
  });

});