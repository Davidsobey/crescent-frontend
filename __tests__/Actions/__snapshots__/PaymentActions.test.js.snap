// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentActions Dispatches the correct action and payload 1`] = `Array []`;
